services/common/src/components/common/ScrollSideMenu.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useEffect } from "react";
+import { withRouter } from "react-router-dom";
+import { Anchor } from "antd";
+import { ISideMenuOption } from "../../interfaces/common/sideMenuOption.interface";
+
+/**
+ * @constant ScrollSideMenu renders react children with an active indicator if the id is in the url.
+ */
+
+interface ScrollSideMenuProps {
+  menuOptions: ISideMenuOption[];
+  featureUrlRoute: any;
+  featureUrlRouteArguments: (string | number)[];
+  history: any;
+  location: any;
+  match: any;
```

Comment:
Would highly recommend taking out history, location, and match as props, using hooks instead. (useHistory, useLocation, useParams). Easier to parse and work with, and less problems converting parent components to TS. You may be able to simplify a lot of the logic where you're doing null checks, etc, on those values.

Reply:
This has been adjusted to use hooks instead of props now.